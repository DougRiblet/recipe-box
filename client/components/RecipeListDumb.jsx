import React from 'react'
import RecipeTitle from './RecipeTitle'

const RecipeListDumb = ({ recipeTitles, changeActiveRecipe }) => (
  <div id='recipe-list-div'>
    <ul>
      {recipeTitles.map(item =>
        <RecipeTitle
          key={item.id}
          title={item.title}
          changeActiveRecipe={() => changeActiveRecipe(item.id)}
        />
      )}
    </ul>
  </div>
)

RecipeListDumb.propTypes = {
  recipeTitles: React.PropTypes.arrayOf(
    React.PropTypes.shape({
      title: React.PropTypes.string.isRequired,
      id: React.PropTypes.string.isRequired
    })
  ),
  changeActiveRecipe: React.PropTypes.func.isRequired
}

export default RecipeListDumb
