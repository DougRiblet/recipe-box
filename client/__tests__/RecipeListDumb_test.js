import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import RecipeListDumb from '../components/RecipeListDumb'

describe('RecipeListDumb', () => {
  test('Snapshot', () => {
    let dummyRecipeTitles = [{id: '101010', title: 'Grilled Cheese Sandwich'}]
    const RecipeListDumb_tree = renderer.create(
      <RecipeListDumb
        recipeTitles={dummyRecipeTitles}
        changeActiveRecipe={() => changeActiveRecipe('010101')}
      />
    ).toJSON()
    expect(RecipeListDumb_tree).toMatchSnapshot()
  })

  test('renders without crashing', () => {
    const div = document.createElement('div')
    let dummyRecipeTitles = [{id: '101010', title: 'Grilled Cheese Sandwich'}]
    ReactDOM.render(<RecipeListDumb
      recipeTitles={dummyRecipeTitles}
      changeActiveRecipe={() => changeActiveRecipe('010101')}
    />, div)
  })

  test('renders component', () => {
    let dummyRecipeTitles = [{id: '101010', title: 'Grilled Cheese Sandwich'}]
    let wrapper = shallow(<RecipeListDumb
      recipeTitles={dummyRecipeTitles}
      changeActiveRecipe={() => changeActiveRecipe('010101')}
    />)
    expect(wrapper.length).toEqual(1)
  })
})
