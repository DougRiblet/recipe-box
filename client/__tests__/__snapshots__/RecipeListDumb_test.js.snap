// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeListDumb Snapshot 1`] = `
<div
  id="recipe-list-div"
>
  <ul>
    <li
      onClick={[Function]}
    >
      Grilled Cheese Sandwich
    </li>
  </ul>
</div>
`;
